platform-sdk/docs/core/tipset-algorithm.md

```diff
@@ -0,0 +1,260 @@
+5 vs 2: 2 < 5, so this is not an advancement
+2 vs 11: 11 > 2, so this is an advancement
+
+The total advancement count from X to Y is therefore 2. Note that the amount an entry advances is not important,
+only if it increased or did not increase.
+```
+
+# Weighted Tipset Advancement Score
+
+A weighted tipset advancement score is very similar to a tipset advancement score. But instead of adding 1 for
+each entry that has a greater value, instead add the amount of consensus weight held by that node.
+
+```
+Example
+
+Suppose an address book has the following weights: A = 5, B = 9, C = 11, D = 2
+
+Find the weighed advancement score between X and Y.
+
+X = [1, 3, 5, 2]
+Y = [7, 3, 2, 11]
+
+The entry for node A advances, add 5. The entry for D advances, add 2. Total weighed advancement score is 7.
+```
+
+# Partial Weighted Tipset Advancement Score
+
+This is similar to the weighed advancement score with one minor tweak. A partial weighted advancement score
+is always calculated from the perspective of a particular node. When computing a partial advancement score,
+ignore the advancement provided by a target node.
+
+```
+Example
+
+Suppose an address book has the following weights: A = 5, B = 9, C = 11, D = 2
+
+Find the partial weighted advancement score between X and Y with a target of node A.
+
+X = [1, 3, 5, 2]
+Y = [7, 3, 2, 11]
+
+The entry for node A advances, but we ignore it because A is the target.
+The entry for D advances, add 2. Total weighed advancement score is 2.
+```
+
+# Snapshot Improvement Scores
+
+Each new event is assigned a tipset improvement score. Each node assigns the events that it creates an
+improvement score. Nodes do not assign events created by other nodes an improvement score.
+
+Keep track of a special tipset, called the "snapshot" tipset. The snapshot starts out empty, i.e.
+`[-1, -1, -1, ..., -1]` at genesis. Periodically, the snapshot is updated to a more recent tipset
+(how this happens is described below).
+
+Each time a node creates a new event, compare that event's tipset to the snapshot tipset, and find the partial
+weighted advancement score of that new event targeting the event's creator.
+
+There is a special threshold that causes the snapshot to advance. The threshold for each node is slightly different,
+unless nodes have the same consensus weight. To calculate the threshold, find the minimum weight required to have >2/3
+of the total weight, and then subtract the node's weight.
+
+```
+Example of computing snapshot advancement threshold
+
+Suppose an address book has the following weights: A = 5, B = 9, C = 11, D = 2. Compute the snapshot advancement
+threshold for node A.
+
+Total consensus weight is 27. The amount of stake required to have >2/3 of the total weight is 19. Subtract A's weight,
+and A's snapshot advancement threshold is 14.
+```
+
+When a new event has a snapshot improvement score that is greater than or equal to the snapshot advancement threshold,
```

Comment:
Here's a thought to take under consideration, but feel free to ignore:

since there is a lot of terminology floating around in this document, perhaps it makes sense to put code ticks around phrases with precise meanings? E.g. 

When a new event has a `snapshot improvement score` that is greater than or equal to the `snapshot advancement threshold`...

Reply:
Or capitalize the first letter of each word in a phrase, and have a list of definitions somewhere for reference

Reply:
I will use the code tics for definitions. Would it be ok if I put these terms in the higher level platform glossary I am working on?